main/realmain.cc

```diff
@@ -753,6 +760,49 @@ int realmain(int argc, char *argv[]) {
             logger->error("Cannot write metrics file at `{}`", opts.metricsFile);
         }
     }
+
+    if (!opts.dumpPackageInfo.empty()) {
+        if (!opts.stripePackages) {
+            logger->error("stripe packages mode needs to be enabled");
+            return 1;
+        }
+        packager::Packager::dumpPackageInfo(*gs, opts.dumpPackageInfo);
+    }
+
+    if (!opts.packageRBIOutput.empty()) {
+        if (opts.stripePackages) {
+            logger->error("Cannot serialize package RBIs in legacy stripe packages mode.");
+            return 1;
+        }
+
+        if (opts.rawInputDirNames.size() != 1) {
+            logger->error("Serializing package RBIs requires one input folder.");
+            return 1;
+        }
+
+        auto relativeIgnorePatterns = opts.relativeIgnorePatterns;
+        auto it = find(relativeIgnorePatterns.begin(), relativeIgnorePatterns.end(), "/__package.rb");
+        if (it != relativeIgnorePatterns.end()) {
+            relativeIgnorePatterns.erase(it);
+        } else {
+            Exception::raise("Couldn't find ignore pattern.");
+        }
+        auto packageFiles = opts.fs->listFilesInDir(opts.rawInputDirNames[0], opts.allowedExtensions, true,
+                                                    opts.absoluteIgnorePatterns, relativeIgnorePatterns);
+        packageFiles.erase(remove_if(packageFiles.begin(), packageFiles.end(), [](const auto &packageFile) {
+            return !absl::EndsWith(packageFile, "__package.rb");
+        }));
+
+        if (!packageFiles.empty()) {
+            auto packageFileRefs = pipeline::reserveFiles(gs, packageFiles);
+            auto packages = pipeline::index(*gs, packageFileRefs, opts, *workers, nullptr);
+            packager::RBIGenerator::run(*gs, move(packages), opts.packageRBIOutput, *workers);
+        } else {
+            logger->error("No package files found!");
+            return 1;
+        }
+    }
```

Comment:
are you sure you want this way down here, after all the metrics have been reported?

seems like we would want this to live up closer to e.g. where the code to handle `--minimize-rbi` lives.

Reply:
Not sure at all; I just plopped this code in what looked like a safe position. Feel free to move.